app/src/main/java/ee/ioc/phon/android/speechutils/ContinuousRawAudioRecorder.java

```diff
@@ -0,0 +1,215 @@
+package ee.ioc.phon.android.speechutils;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import ee.ioc.phon.android.speechutils.utils.AudioUtils;
+
+public class ContinuousRawAudioRecorder extends AbstractAudioRecorder {
+
+    private static final int DEFAULT_BUFFER_LENGTH_IN_MILLIS = 2000;
+    private static final String LOG_FILTER = "continuous-recorder: ";
+
+    private SessionStartPointer mSessionStartPointer = SessionStartPointer.beginningOfBuffer();
+    private AtomicBoolean mRecordingToFile = new AtomicBoolean(false);
+
```

Comment:
This is my solution to - from what moment in time should I start the consume process?
By default, this class defines a buffer of 2 seconds (as defined in DEFAULT_BUFFER_LENGTH_IN_MILLIS above). For handling the case of synchronization between a keyword detection (on DSP) and STT processing in the Android side, we should be able to stream the buffer from a certain point in the past (assuming that the trigger is handled in the Android side a few/a lot of millis after the real world end of keyword). So we should keep a big enough buffer (2 seconds sounds reasonable) and start the consumption from (now - TBD).
My way of marking this TBD while keeping the original API (consumeRecording without parameters) is to tell the recorder where should be the next session's starting point.
There are a few options with this "marker" object - now, from the beginning of the buffer and now - X where X can be in seconds/millis.